test/v1/revision.go

```diff
@@ -75,6 +75,14 @@ func IsRevisionReady(r *v1.Revision) (bool, error) {
 	return r.Generation == r.Status.ObservedGeneration && r.Status.IsReady(), nil
 }
 
+// IsRevisionPinned will check if the revision is pinned to a route.
+func IsRevisionPinned(r *v1.Revision) (bool, error) {
```

Comment:
It seems that if you just return false that will percolate nicely without error? (return CheckRevisionState)

Reply:
Yeah. That's right. I just wanted to align the function signature with the other functions of this type so that they can be used similarly.

Reply:
Signature is fine. But returning `false` rather than error would achieve same benefit, no?

Reply:
Yes. I meant signature including return parameters/type. Returning false would be simpler in this case indeed.

Reply:
Ehmm. Actually, the CheckRevisionState function expects the signature I have now.

Reply:
Yes, don't change the signature, change the logic :)

Reply:
Oooh, ok. Do you mean something like this?
```
// IsRevisionPinned will check if the revision is pinned to a route.
func IsRevisionPinned(r *v1beta1.Revision) (bool, error) {
	_, pinned := r.Annotations[serving.RevisionLastPinnedAnnotationKey]
	return pinned, nil
}
```

Reply:
Yep.

Reply:
And in the call site just
`return X(...)`, rather than `if _, err := X(...); err != nil...`.

Reply:
Pushed changes. Unfortunately, the last suggestion from you is not possible because I'm not directly calling IsRevisionPinned but I'm passing it to CheckRevisionState which, itself returns only `error`. But the enclosing function expects again (bool, error). So there's double conversion of return types throughout the method calls.